Mathlib/Algebra/Category/ModuleCat/Basic.lean

```diff
@@ -174,70 +235,37 @@ def ofSelfIso (M : ModuleCat R) : ModuleCat.of R M ≅ M where
   inv := 𝟙 M
 
 theorem isZero_of_subsingleton (M : ModuleCat R) [Subsingleton M] : IsZero M where
-  unique_to X := ⟨⟨⟨(0 : M →ₗ[R] X)⟩, fun f => by
+  unique_to X := ⟨⟨⟨asHom (0 : M →ₗ[R] X)⟩, fun f => by
     ext x
     rw [Subsingleton.elim x (0 : M)]
     dsimp
     simp⟩⟩
-  unique_from X := ⟨⟨⟨(0 : X →ₗ[R] M)⟩, fun f => by
+  unique_from X := ⟨⟨⟨asHom (0 : X →ₗ[R] M)⟩, fun f => by
     ext x
     subsingleton⟩⟩
 
 instance : HasZeroObject (ModuleCat.{v} R) :=
   ⟨⟨of R PUnit, isZero_of_subsingleton _⟩⟩
 
-variable {M N U : ModuleCat.{v} R}
-
-@[simp]
-theorem id_apply (m : M) : (𝟙 M : M → M) m = m :=
-  rfl
-
-@[simp]
-theorem coe_comp (f : M ⟶ N) (g : N ⟶ U) : (f ≫ g : M → U) = g ∘ f :=
-  rfl
-
-theorem comp_def (f : M ⟶ N) (g : N ⟶ U) : f ≫ g = g.comp f :=
-  rfl
-
-@[simp] lemma forget_map (f : M ⟶ N) : (forget (ModuleCat R)).map f = (f : M → N) := rfl
-
 end ModuleCat
 
 variable {R}
 variable {X₁ X₂ : Type v}
 
 open ModuleCat
 
-/-- Reinterpreting a linear map in the category of `R`-modules. -/
-def ModuleCat.asHom [AddCommGroup X₁] [Module R X₁] [AddCommGroup X₂] [Module R X₂] :
-    (X₁ →ₗ[R] X₂) → (ModuleCat.of R X₁ ⟶ ModuleCat.of R X₂) :=
-  id
-
-@[deprecated (since := "2024-10-06")] alias ModuleCat.ofHom := ModuleCat.asHom
-
 /-- Reinterpreting a linear map in the category of `R`-modules -/
 scoped[ModuleCat] notation "↟" f:1024 => ModuleCat.asHom f
 
-@[simp 1100]
-theorem ModuleCat.asHom_apply {R : Type u} [Ring R] {X Y : Type v} [AddCommGroup X] [Module R X]
-    [AddCommGroup Y] [Module R Y] (f : X →ₗ[R] Y) (x : X) : (↟ f) x = f x :=
-  rfl
-
 @[deprecated (since := "2024-10-06")] alias ModuleCat.ofHom_apply := ModuleCat.asHom_apply
 
-/-- Reinterpreting a linear map in the category of `R`-modules. -/
-def ModuleCat.asHomRight [AddCommGroup X₁] [Module R X₁] {X₂ : ModuleCat.{v} R} :
-    (X₁ →ₗ[R] X₂) → (ModuleCat.of R X₁ ⟶ X₂) :=
-  id
+-- Since `of` and the coercion now roundtrip reducibly, we don't need to distinguish in which place
+-- we need to add `of` when coercing from linear maps to morphisms.
+@[deprecated ModuleCat.asHom (since := "2024-11-29")] alias ModuleCat.asHomRight := ModuleCat.asHom
+@[deprecated ModuleCat.asHom (since := "2024-11-29")] alias ModuleCat.asHomLeft := ModuleCat.asHom
 
 /-- Reinterpreting a linear map in the category of `R`-modules. -/
 scoped[ModuleCat] notation "↾" f:1024 => ModuleCat.asHomRight f
-
-/-- Reinterpreting a linear map in the category of `R`-modules. -/
-def ModuleCat.asHomLeft {X₁ : ModuleCat.{v} R} [AddCommGroup X₂] [Module R X₂] :
-    (X₁ →ₗ[R] X₂) → (X₁ ⟶ ModuleCat.of R X₂) :=
-  id
-
 /-- Reinterpreting a linear map in the category of `R`-modules. -/
 scoped[ModuleCat] notation "↿" f:1024 => ModuleCat.asHomLeft f
```

Comment:
These are implicitly deprecated, right? Let's write something in the doc-string.

Reply:
Indeed, since this notation resolves to the deprecated `asHomLeft`/`asHomRight`, we get a deprecation warning if we actually try to use the notation. I'll add the same warning to the docstring for the notation itself.